### PR TITLE
Remove `Dimension::Undefined`

### LINKED ISSUE #114: Consider removing Dimension::Undefined
My instincts suggest this should be an `Option<Dimension>`.

### PR DESCRIPTION
**Pull request by Weibye**
# Objective

Fixes #114

Replaces #160 and #148

## Context

The new Resolve-traits really helped clean this up :)

## Feedback wanted

- I'm not yet sure why the `simple_child` test are failing

### CODE REVIEW

File: tests/simple_child.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ fn simple_child() {
     let node1 = taffy
         .new_with_children(
             taffy::style::FlexboxLayout {
-                size: taffy::geometry::Size { width: Dimension::Undefined, height: Dimension::Undefined },
+                size: taffy::geometry::Size { width: Dimension::Auto, height: Dimension::Auto },
```
Comment (Weibye) [inline tests/simple_child.rs:48]: This should be `None` instead of `Auto.` Which means I need to change `Size` to be `Option<Dimension>` here as well

**Comment by Weibye** [comment]
I suspect the current failures comes from Measure functions assuming `Size<Dimension>` and not `Size<Option<Dimension>>` so that is what I will investigate next

**Comment by Weibye** [comment]
The generated files are still showing in the review. Is #149 not working as intended?

**Comment by Weibye** [comment]
🤔 Tests fails on height calculation but all width calculations are correct. Trying to dig into why that is but progress is slow (lots of new things to learn)

**Comment by alice-i-cecile** [comment]
Going to close this out: this will be a pain to rebase. Anyone can feel free to remake this though!

**Comment by Weibye** [comment]
Yeah, it was a worthwhile attempt but I couldn't get it across the finish line due to some nuances in the algorithm I could not fully figure out. Now with taffy undergoing some great improvements I think this could be easier to attempt again after #238 and #246 should it still prove useful to do.

**Pull request closed without merging**